src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionReader.cs

```diff
@@ -33,7 +33,15 @@ public object ReflectionReadClass(XmlReaderDelegator xmlReader, XmlObjectSeriali
             context.AddNewObject(obj);
             InvokeOnDeserializing(context, classContract, obj);
 
-            ReflectionReadMembers(xmlReader, context, memberNames, memberNamespaces, classContract, ref obj);
+            if (classContract.IsISerializable)
+            {
+                obj = ReadISerializable(xmlReader, context, classContract);
```

Comment:
I opened https://github.com/dotnet/corefx/issues/14651 for tracking potential issues with reading ISerializable.